tests/components/template/test_binary_sensor.py

```diff
@@ -221,52 +218,27 @@ def test_match_all(self, _async_render):
                     "binary_sensor": {
                         "platform": "template",
                         "sensors": {
-                            "match_all_template_sensor": {"value_template": "{{ 42 }}"}
+                            "match_all_template_sensor": {
+                                "value_template": (
+                                    "{% for state in states %}"
+                                    "{% if state.entity_id == 'sensor.humidity' %}"
+                                    "{{ state.entity_id }}={{ state.state }}"
+                                    "{% endif %}"
+                                    "{% endfor %}"
+                                ),
+                            },
                         },
                     }
                 },
             )
 
-        self.hass.block_till_done()
         self.hass.start()
         self.hass.block_till_done()
-        init_calls = len(_async_render.mock_calls)
+        init_calls = len(_update_state.mock_calls)
 
         self.hass.states.set("sensor.any_state", "update")
         self.hass.block_till_done()
-        assert len(_async_render.mock_calls) == init_calls
-
-    def test_attributes(self):
-        """Test the attributes."""
-        vs = run_callback_threadsafe(
-            self.hass.loop,
-            template.BinarySensorTemplate,
-            self.hass,
-            "parent",
-            "Parent",
-            "motion",
-            template_hlpr.Template("{{ 1 > 1 }}", self.hass),
-            None,
-            None,
-            None,
-            MATCH_ALL,
-            None,
-            None,
-            None,
-            None,
-        ).result()
-        assert not vs.should_poll
-        assert "motion" == vs.device_class
-        assert "Parent" == vs.name
-
-        run_callback_threadsafe(self.hass.loop, vs.async_check_state).result()
-        assert not vs.is_on
-
-        # pylint: disable=protected-access
-        vs._template = template_hlpr.Template("{{ 2 > 1 }}", self.hass)
-
-        run_callback_threadsafe(self.hass.loop, vs.async_check_state).result()
-        assert vs.is_on
+        assert len(_update_state.mock_calls) == init_calls
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6042836)